src/main/groovy/wooga/gradle/hockey/tasks/HockeyUploadTask.groovy

```diff
@@ -91,17 +113,33 @@ class HockeyUploadTask extends ConventionTask {
 
     @TaskAction
     protected void upload() {
-        project.exec {
-            executable "curl"
-
-            // TODO: make status and notify configurable, add changelog support
-            args "-F", "status=2"
-            args "-F", "notify=2"
-            args "-F", "ipa=@" + getBinary()
-            args "-H", "X-HockeyAppToken: " + getApiToken()
-            args "https://rink.hockeyapp.net/api/2/apps/" + getApplicationIdentifier() + "/app_versions/upload"
-            args "-v", "-f"
+        HttpClient client = HttpClientBuilder.create().build()
+        HttpPost post = new HttpPost("https://rink.hockeyapp.net/api/2/apps/${getApplicationIdentifier()}/app_versions/upload")
+
+        post.setHeader("X-HockeyAppToken", getApiToken())
+
+        FileBody binary = new FileBody(getBinary())
+
+        // TODO: make status and notify configurable, add changelog support
+        HttpEntity content = MultipartEntityBuilder.create()
+                .addPart("ipa", binary)
+                .addTextBody("status", "2")
+                .addTextBody("notify", "2")
+                .build()
+
+        post.setEntity(content)
+        HttpResponse response = client.execute(post)
+
+        if(response.statusLine.statusCode != 201) {
+            throw new GradleException("unable to upload to hockey")
```

Comment:
coe be more specific, maybe add the returned status code to the error message? or will it be exposed at some other point?

Reply:
At this point I only wanted to move from `curl` to `httpclient`. Curl will fail with exit code > 0 for anything else than a result in the `200` area. And the actual message we would see in gradle was/is limited. What I mean is: `Error handling and proper logging` was not my focus here. I just want to fail, when the result is not a `201` which is the documented success return from Hockey, to have the same basic behavior has the `curl` solution.